File: schemaorggo/deliverymethod.go
package schemaorggo

import "encoding/json"

// DeliveryMethod see : https://schema.org/DeliveryMethod
type DeliveryMethod struct {
	Enumeration

	typeContext

	// SupersededBy see : http://meta.schema.org/supersededBy
	// Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
	// types : Class Enumeration Property
	SupersededBy interface{} `json:"supersededBy,omitempty"`
}

func (v DeliveryMethod) MarshalJSONWithTypeContext() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "DeliveryMethod"

	return json.Marshal(v)
}

func (v *DeliveryMethod) MarshalJSON() ([]byte, error) {
	if v == nil {
		return []byte("null"), nil
	}

	v.C = "http://schema.org"
	v.T = "DeliveryMethod"

	return json.Marshal(*v)
}

File: schemaorgphp/classes/delivery_method.php
<?php

class DeliveryMethod extends Enumeration implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'DeliveryMethod';
	
	/**
	 * Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
	 * see : https://meta.schema.org/supersededBy
	 * @var \Class|\Class[]|\Enumeration|\Enumeration[]|\Property|\Property[]
	 */
	public var $superseded_by;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'DeliveryMethod'
		);
		
		$serialized = so_json_serialize( $this->superseded_by );
		if ( ! empty( $serialized ) ) {
			$out['supersededBy'] = $serialized;
		}
		
		return $out;
	}
}
